samples/bluetooth/fast_pair/locator_tag/src/motion_detector/platform_thingy.c

```diff
@@ -0,0 +1,235 @@
+	LOG_INF("FMDN: motion detector period expired. Reporting that the motion was %sdetected",
+		is_motion_detected ? "" : "not ");
+	memset(motion_detector.rot, 0, sizeof(motion_detector.rot));
+	return is_motion_detected;
+}
+
+static void fmdn_motion_detector_stop(void)
+{
+	/* It is assumed that this function executes in the cooperative thread context. */
+	__ASSERT_NO_MSG(!k_is_preempt_thread());
+	__ASSERT_NO_MSG(!k_is_in_isr());
+
+	__ASSERT_NO_MSG(motion_detector.active);
+	LOG_INF("FMDN: motion detector stopped");
+	motion_detector.active = false;
+	memset(motion_detector.rot, 0, sizeof(motion_detector.rot));
+	app_ui_state_change_indicate(APP_UI_STATE_MOTION_DETECTOR_ACTIVE, motion_detector.active);
+}
+
+static const struct bt_fast_pair_fmdn_motion_detector_cb fmdn_motion_detector_cb = {
+	.start = fmdn_motion_detector_start,
+	.period_expired = fmdn_motion_detector_period_expired,
+	.stop = fmdn_motion_detector_stop,
+};
+
+static int sensor_configure(const struct device *sensor)
+{
+	struct sensor_value scale;
+	struct sensor_value sampling_freq;
+	struct sensor_value oversampling;
+	int err;
+
+	SENSOR_VALUE_SCALE_1000_DEG_PER_SEC_SET(scale);
+	err = sensor_attr_set(sensor, SENSOR_CHAN_GYRO_XYZ, SENSOR_ATTR_FULL_SCALE, &scale);
+	if (err) {
+		return err;
+	}
+
+	SENSOR_VALUE_OVERSAMPLING_DISABLE(oversampling);
+	err = sensor_attr_set(sensor, SENSOR_CHAN_GYRO_XYZ, SENSOR_ATTR_OVERSAMPLING,
+			      &oversampling);
+	if (err) {
+		return err;
+	}
+
+	SENSOR_VALUE_SAMPLING_FREQ_25_HZ_SET(sampling_freq);
+	err = sensor_attr_set(sensor, SENSOR_CHAN_GYRO_XYZ,
+			      SENSOR_ATTR_SAMPLING_FREQUENCY,
+			      &sampling_freq);
+	if (err) {
+		return err;
+	}
+
+	return err;
+}
+
+static void gyro_poll_thread_fn(void *dummy0, void *dummy1, void *dummy2)
+{
+	ARG_UNUSED(dummy0);
+	ARG_UNUSED(dummy1);
+	ARG_UNUSED(dummy2);
+
+	while (true) {
+		k_spinlock_key_t key;
+		struct sensor_value val[ARRAY_SIZE(motion_detector.rot)];
+
+		k_sem_take(&gyro_poll_sem, K_FOREVER);
+
+		sensor_sample_fetch(motion_detector.dev);
+		sensor_channel_get(motion_detector.dev, SENSOR_CHAN_GYRO_XYZ, val);
+
+		key = k_spin_lock(&motion_detector.lock);
+		if (motion_detector.active) {
+			for (size_t i = 0; i < ARRAY_SIZE(motion_detector.rot); i++) {
+				/* Rotation = Angular velocity * Time = Angular velocity /
+				 *					Sampling frequency.
+				 */
+				motion_detector.rot[i] += sensor_value_to_double(&val[i]) /
+							  GYRO_SAMPLING_FREQ;
+			}
+			k_sem_give(&gyro_poll_sem);
+		}
+		k_spin_unlock(&motion_detector.lock, key);
+
+		k_msleep(MSEC_PER_SEC / GYRO_SAMPLING_FREQ);
+	}
+}
+
+K_THREAD_DEFINE(gyro_poll_thread, GYRO_POLL_THREAD_STACK_SIZE, gyro_poll_thread_fn,
+		NULL, NULL, NULL, GYRO_POLL_THREAD_PRIORITY, 0, 0);
+
+int app_motion_detector_init(void)
+{
+	const struct gpio_dt_spec pwr = GPIO_DT_SPEC_GET(GYRO_PWR_NODE, enable_gpios);
+	int err;
+
+	if (!device_is_ready(pwr.port)) {
+		LOG_ERR("FMDN: GYRO_PWR_NODE is not ready");
```

Comment:
We are not directly interacting in this logs with the FMDN, consider using other prefix or none at all, similarly in logs below in this function (maybe beside bt_fast_pair_fmdn_motion_detector_cb_register as it interacts with it :D)